be/src/formats/orc/column_reader.cpp

```diff
@@ -0,0 +1,1060 @@
+        const TypeDescriptor& key_type = type.children[0];
+        const TypeDescriptor& value_type = type.children[1];
+        if (key_type.is_unknown_type()) {
+            child_readers.emplace_back(nullptr);
+        } else {
+            ASSIGN_OR_RETURN(
+                    std::unique_ptr<ORCColumnReader> key_reader,
+                    ORCColumnReader::create(key_type, orc_mapping->get_orc_type_child_mapping(0).orc_type, true,
+                                            orc_mapping->get_orc_type_child_mapping(0).orc_mapping, reader));
+            child_readers.emplace_back(std::move(key_reader));
+        }
+
+        if (value_type.is_unknown_type()) {
+            child_readers.emplace_back(nullptr);
+        } else {
+            ASSIGN_OR_RETURN(
+                    std::unique_ptr<ORCColumnReader> value_reader,
+                    ORCColumnReader::create(value_type, orc_mapping->get_orc_type_child_mapping(1).orc_type, true,
+                                            orc_mapping->get_orc_type_child_mapping(1).orc_mapping, reader));
+            child_readers.emplace_back(std::move(value_reader));
+        }
+        return std::make_unique<MapColumnReader>(type, orc_type, nullable, reader, std::move(child_readers));
+    }
+    default:
+        return Status::InternalError("Unsupported type");
+    }
+}
+
+Status BooleanColumnReader::get_next(orc::ColumnVectorBatch* cvb, ColumnPtr& col, size_t from, size_t size) {
+    if (_nullable) {
+        auto* data = down_cast<orc::LongVectorBatch*>(cvb);
+        int col_start = col->size();
+        col->resize(col->size() + size);
+
+        auto c = ColumnHelper::as_raw_column<NullableColumn>(col);
+        auto* nulls = c->null_column()->get_data().data();
+        auto* values = ColumnHelper::cast_to_raw<TYPE_BOOLEAN>(c->data_column())->get_data().data();
+
+        auto* cvbd = data->data.data();
+        auto pos = from;
+        if (cvb->hasNulls) {
+            auto* cvbn = reinterpret_cast<uint8_t*>(cvb->notNull.data());
+            for (int i = col_start; i < col_start + size; ++i, ++pos) {
+                nulls[i] = !cvbn[pos];
+            }
+        }
+        pos = from;
+        for (int i = col_start; i < col_start + size; ++i, ++pos) {
+            values[i] = (cvbd[pos] != 0);
+        }
+        c->update_has_null();
+    } else {
+        auto* data = down_cast<orc::LongVectorBatch*>(cvb);
+
+        int col_start = col->size();
+        col->resize(col_start + size);
+
+        auto* values = ColumnHelper::cast_to_raw<TYPE_BOOLEAN>(col)->get_data().data();
+
+        auto* cvbd = data->data.data();
+        for (int i = col_start; i < col_start + size; ++i, ++from) {
+            values[i] = (cvbd[from] != 0);
+        }
+    }
+    return Status::OK();
+}
+
+template <LogicalType Type>
+Status IntColumnReader<Type>::get_next(orc::ColumnVectorBatch* cvb, ColumnPtr& col, size_t from, size_t size) {
+    if (_nullable) {
```

Comment:
abstract duplicated block of  if.. else..

Reply:
refactor it step by step